litestar/contrib/jwt/jwt_token.py

```diff
@@ -63,8 +66,8 @@ def __post_init__(self) -> None:
         else:
             raise ImproperlyConfiguredException("iat must be a current or past time")
 
-    @staticmethod
-    def decode(encoded_token: str, secret: str | dict[str, str], algorithm: str) -> Token:
+    @classmethod
+    def decode(cls, encoded_token: str, secret: str | dict[str, str], algorithm: str) -> Token:
```

Comment:
```suggestion
    def decode(cls, encoded_token: str, secret: str | dict[str, str], algorithm: str) -> Self:
```

Import `Self` from `typing_extensions`. The reason this is important is that if someone subclasses `Token`, the current implementation will not type this method as returning an instance of the subtype, even though it does.